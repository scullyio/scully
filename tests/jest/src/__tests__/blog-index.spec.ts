import { readPage, replaceIndexNG } from '../test-config.helper';
import { expect } from '@jest/globals';

describe('Static: Test blog index', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readPage('blog');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});

describe('ContentFolder: Test blog/page-1', () => {
  it('Check contentPlugin render', () => {
    const index: string = readPage('blog/page-1');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});

describe('ContentFolder: Test blog/page-6 expired publishDate', () => {
  it('Check contentPlugin render', () => {
    const index: string = readPage('blog/page-6');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});

describe('ContentFolder: Test foreign slug', () => {
  it('Check contentPlugin render', () => {
    const index: string = readPage('blog/Русский-слаг');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});
