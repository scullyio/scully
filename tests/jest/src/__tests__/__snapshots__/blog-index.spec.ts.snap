// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentFolder: Test blog/page-1 Check contentPlugin render 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      sample-blog
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
      h1[]{background-color:#4169e1;color:#fff;width:100%;text-align:center;padding:5px;border-radius:8px;margin-top:.4rem;margin-bottom:.4rem}
    </style>
    <style>
      :host {
        display: none;
      }
      scully-content {
        display: none;
      }
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-blog-holder>
          <h3>
            Scully blog content
          </h3>
          <hr>
          <router-outlet>
          </router-outlet>
          <app-blog ="">
            <h1 id="page-1"
                =""
            >
              Page 1
            </h1>
            <h2 id="or-how-to-do-interesting-blog-things"
                =""
            >
              or, how to do interesting blog things
            </h2>
            <pre class="language-typescript"
                 =""
            >
              <code class="language-typescript"
                    =""
              >
                <span class="token builtin"
                      =""
                >
                  console
                </span>
                <span class="token punctuation"
                      =""
                >
                  .
                </span>
                <span class="token function"
                      =""
                >
                  log
                </span>
                <span class="token punctuation"
                      =""
                >
                  (
                </span>
                <span class="token string"
                      =""
                >
                  'hello world'
                </span>
                <span class="token punctuation"
                      =""
                >
                  )
                </span>
                <span class="token punctuation"
                      =""
                >
                  ;
                </span>
                <span class="token builtin"
                      =""
                >
                  console
                </span>
                <span class="token punctuation"
                      =""
                >
                  .
                </span>
                <span class="token function"
                      =""
                >
                  log
                </span>
                <span class="token punctuation"
                      =""
                >
                  (
                </span>
                <span class="token string"
                      =""
                >
                  'yah'
                </span>
                <span class="token punctuation"
                      =""
                >
                  )
                </span>
                <span class="token punctuation"
                      =""
                >
                  ;
                </span>
                <span class="token keyword"
                      =""
                >
                  const
                </span>
                x
                <span class="token operator"
                      =""
                >
                  =
                </span>
                <span class="token number"
                      =""
                >
                  123
                </span>
                <span class="token punctuation"
                      =""
                >
                  ;
                </span>
              </code>
            </pre>
            <p ="">
              Hello Dolly?
            </p>
            <p ="">
              Related information
              <a href="/blog/___UNPUBLISHED___k5nhcflm_SJwD4Z0QDrIHg1PGHo2mrfLZE8sfUsPy"
                 =""
              >
                page-2
              </a>
            </p>
            <p ="">
              <a href="/content/two"
                 =""
              >
                content page
              </a>
            </p>
            <h1 id="down-under"
                =""
            >
              Down-under
            </h1>
            <p ="">
              <a href="/home"
                 =""
              >
                site-map
              </a>
            </p>
            <scully-content ="">
            </scully-content>
          </app-blog>
          <hr>
          <h4>
            End of blog content
          </h4>
          <span>
            <a href="/blog/2014/2/12/asdf">
              /blog/2014/2/12/asdf
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/%D0%A0%D1%83%D1%81%D1%81%D0%BA%D0%B8%D0%B9-%D1%81%D0%BB%D0%B0%D0%B3">
              /blog/–†—É—Å—Å–∫–∏–π-—Å–ª–∞–≥
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-1">
              /blog/page-1
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-3">
              /blog/page-3
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-4">
              /blog/page-4
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-5">
              /blog/page-5
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-6">
              /blog/page-6
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/test/testing">
              /blog/test/testing
            </a>
            <br>
          </span>
        </app-blog-holder>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;

exports[`ContentFolder: Test blog/page-6 expired publishDate Check contentPlugin render 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      sample-blog
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
      h1[]{background-color:#4169e1;color:#fff;width:100%;text-align:center;padding:5px;border-radius:8px;margin-top:.4rem;margin-bottom:.4rem}
    </style>
    <style>
      :host {
        display: none;
      }
      scully-content {
        display: none;
      }
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-blog-holder>
          <h3>
            Scully blog content
          </h3>
          <hr>
          <router-outlet>
          </router-outlet>
          <app-blog ="">
            <h1 id="page-6"
                =""
            >
              Page 6
            </h1>
            <p ="">
              Navigate to
              <a href="javascript:;"
                 =""
                 data-hash="end-of-page"
                 onclick="document.location.hash=this.dataset.hash"
              >
                end of page
              </a>
            </p>
            <ul ="">
              <li ="">
                1
              </li>
              <li ="">
                2
              </li>
              <li ="">
                3
              </li>
              <li ="">
                4
              </li>
              <li ="">
                5
              </li>
              <li ="">
                5
              </li>
              <li ="">
                6
              </li>
              <li ="">
                7
              </li>
              <li ="">
                8
              </li>
              <li ="">
                9
              </li>
              <li ="">
                10
              </li>
              <li ="">
                11
              </li>
              <li ="">
                12
              </li>
              <li ="">
                13
              </li>
              <li ="">
                14
              </li>
              <li ="">
                15
              </li>
              <li ="">
                15
              </li>
              <li ="">
                16
              </li>
              <li ="">
                17
              </li>
              <li ="">
                18
              </li>
              <li ="">
                19
              </li>
              <li ="">
                20
              </li>
              <li ="">
                21
              </li>
              <li ="">
                22
              </li>
              <li ="">
                23
              </li>
              <li ="">
                24
              </li>
              <li ="">
                25
              </li>
              <li ="">
                25
              </li>
              <li ="">
                26
              </li>
              <li ="">
                27
              </li>
              <li ="">
                28
              </li>
              <li ="">
                29
              </li>
              <li ="">
                30
              </li>
            </ul>
            <h2 id="this-page-is-auto-published-after-2020-01-02"
                =""
            >
              This page is auto published after 2020-01-02
            </h2>
            <p ="">
              Related information
              <a href="/blog/page-5"
                 =""
              >
                page-5
              </a>
            </p>
            <p ="">
              <a href="/home"
                 =""
              >
                site-map
              </a>
            </p>
            <h2 id="end-of-page"
                =""
            >
              End of page
            </h2>
            <p ="">
              <a href="javascript:;"
                 =""
                 data-hash="page-6"
                 onclick="document.location.hash=this.dataset.hash"
              >
                top
              </a>
            </p>
            <scully-content ="">
            </scully-content>
          </app-blog>
          <hr>
          <h4>
            End of blog content
          </h4>
          <span>
            <a href="/blog/2014/2/12/asdf">
              /blog/2014/2/12/asdf
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/%D0%A0%D1%83%D1%81%D1%81%D0%BA%D0%B8%D0%B9-%D1%81%D0%BB%D0%B0%D0%B3">
              /blog/–†—É—Å—Å–∫–∏–π-—Å–ª–∞–≥
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-1">
              /blog/page-1
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-3">
              /blog/page-3
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-4">
              /blog/page-4
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-5">
              /blog/page-5
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-6">
              /blog/page-6
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/test/testing">
              /blog/test/testing
            </a>
            <br>
          </span>
        </app-blog-holder>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;

exports[`ContentFolder: Test foreign slug Check contentPlugin render 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      sample-blog
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
      h1[]{background-color:#4169e1;color:#fff;width:100%;text-align:center;padding:5px;border-radius:8px;margin-top:.4rem;margin-bottom:.4rem}
    </style>
    <style>
      :host {
        display: none;
      }
      scully-content {
        display: none;
      }
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-blog-holder>
          <h3>
            Scully blog content
          </h3>
          <hr>
          <router-outlet>
          </router-outlet>
          <app-blog ="">
            <h1 id="this-is-my-post-with-a-slug-in-russian"
                =""
            >
              This is my post with a slug in Russian
            </h1>
            <p ="">
              the slug should read: –†—É—Å—Å–∫–∏–π-—Å–ª–∞–≥
            </p>
            <scully-content ="">
            </scully-content>
          </app-blog>
          <hr>
          <h4>
            End of blog content
          </h4>
          <span>
            <a href="/blog/2014/2/12/asdf">
              /blog/2014/2/12/asdf
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/%D0%A0%D1%83%D1%81%D1%81%D0%BA%D0%B8%D0%B9-%D1%81%D0%BB%D0%B0%D0%B3">
              /blog/–†—É—Å—Å–∫–∏–π-—Å–ª–∞–≥
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-1">
              /blog/page-1
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-3">
              /blog/page-3
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-4">
              /blog/page-4
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-5">
              /blog/page-5
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/page-6">
              /blog/page-6
            </a>
            <br>
          </span>
          <span>
            <a href="/blog/test/testing">
              /blog/test/testing
            </a>
            <br>
          </span>
        </app-blog-holder>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;

exports[`Static: Test blog index Check clean blog index by scully 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      sample-blog
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
      []{display:grid;grid-template-columns:repeat(3,1fr);gap:10px}[]   h1[]:first-child{grid-column:span 3;text-align:center}article[]{background-color:#4169e1;padding:5px;border-radius:5px;color:#f5f5f5}
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home/"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-blog-list ="">
          <h1 ="">
            Overview of blog posts
          </h1>
          <a =""
             href="/blog/2014/2/12/asdf/"
          >
            <article ="">
              <h3 ="">
                Deeply embedded
              </h3>
              <h4 ="">
                11/27/19
              </h4>
              <p ="">
                This is the deep folder demo page in this sample.
              </p>
            </article>
          </a>
          <a =""
             href="/blog/%D0%A0%D1%83%D1%81%D1%81%D0%BA%D0%B8%D0%B9-%D1%81%D0%BB%D0%B0%D0%B3"
          >
            <article ="">
              <h3 ="">
                This is my post in Russian
              </h3>
              <h4 ="">
              </h4>
              <p ="">
                blog description
              </p>
            </article>
          </a>
          <a =""
             href="/blog/page-1/"
          >
            <article ="">
              <h3 ="">
                My first page
              </h3>
              <h4 ="">
                11/26/19
              </h4>
              <p ="">
                This is the first demo page in this sample.
              </p>
            </article>
          </a>
          <a =""
             href="/blog/page-3/"
          >
            <article ="">
              <h3 ="">
                My third page
              </h3>
              <h4 ="">
                11/28/19
              </h4>
              <p ="">
                At this point, I should write something different in here.
              </p>
            </article>
          </a>
          <a =""
             href="/blog/page-4/"
          >
            <article ="">
              <h3 ="">
                The Rainbow Cat, A free book out of the Guttenberg library
              </h3>
              <h4 ="">
                12/16/19
              </h4>
              <p ="">
              </p>
            </article>
          </a>
          <a =""
             href="/blog/page-5/"
          >
            <article ="">
              <h3 ="">
                /blog/page-5
              </h3>
              <h4 ="">
              </h4>
              <p ="">
              </p>
            </article>
          </a>
          <a =""
             href="/blog/page-6/"
          >
            <article ="">
              <h3 ="">
                My sixth page
              </h3>
              <h4 ="">
                1/1/20
              </h4>
              <p ="">
                This is the sixth demo page in this sample.
              </p>
            </article>
          </a>
          <a =""
             href="/blog/test/testing/"
          >
            <article ="">
              <h3 ="">
                My test sub page
              </h3>
              <h4 ="">
                11/27/19
              </h4>
              <p ="">
                This is the second demo page in this sample.
              </p>
            </article>
          </a>
        </app-blog-list>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;
